### DISCUSSION TITLE
Mass updated titles via sql - how to trigger sync?

### DISCUSSION THREAD

**Discussion by contributor** [TriliumNext/Trilium#3586] [General]
Hello

I updated note titles directly in the document.db using sql on Windows. The changes are reflected in Windows client correctly, but they are not synced to Web instance (even after calling Options>Advanced>Force Full Sync). I realize, that Trilium doesn't know about changes made directly in the database so this the reason of sync problem. Any way to trigger sync? What would be the proper way to update notes in the database via sql so sync works automatically? Update modified date?

**Comment by zadam** [comment]
There's a separate table to keep track of the sync process - `entity_changes`, you would have to update `hash` and `utcDateChanged` columns for all affected entities. It's a better idea to use JS backend script notes to do such changes since they will do all necessary "book keeping" for you.

In your case, you can click "Fill entity changes records" which will regenerate them and thus force sync. The side where this is triggered will be newer and thus "stronger" to overwrite the other instance data (since they are now out of sync):

![image](https://user-images.githubusercontent.com/617641/216453049-7417efb1-db33-4d3a-bf11-bd1e0e1b8081.png)

**Comment by contributor** [comment]
Thank you, good to know. On my db (or Trilium version?) "Fill entity changes records" crashes
```
13:25:35.879 DB size: 243466 KB
13:25:36.923 Trusted reverse proxy: false
13:25:36.923 App HTTP server starting up at port 37777
13:25:36.923 {
  "appVersion": "0.58.7",
  "dbVersion": 212,
  "syncVersion": 29,
  "buildDate": "2023-01-17T23:14:58+01:00",
  "buildRevision": "a3149aecf41bac3c559ebbd1865e916264985ac3",
  "dataDirectory": "****",
  "clipperProtocolVersion": "1.0",
  "utcDateTime": "2023-02-03T11:25:35.878Z"
}

....

13:25:59.572 ERROR: post /api/sync/fill-entity-changes threw exception: 'Cannot read properties of undefined (reading 'generateHash')', stack: TypeError: Cannot read properties of undefined (reading 'generateHash')
    at Function.<anonymous> (..Trilium\resources\app.asar\src\services\entity_changes.js:108:34)
```

I then executed "DELETE FROM entity_changes". Fill changes crashes with the same error.

**Reply by contributor** [reply]
Looks like ""DELETE FROM entity_changes" did trigger sync, as my other instance received the changes. Though ""Fill entity changes records" still crashes.
